tests/test_app_integration.py

```diff
@@ -0,0 +1,186 @@
+
+    return MockResponse(None, 404)
+
+def mock_post_requests(*args, **kwargs):
+    class MockResponse:
+        def __init__(self, json_data, status_code):
+            self.json_data = json_data
+            self.status_code = status_code
+
+        def json(self):
+            return self.json_data
+
+    if args[0] == 'https://api.trello.com/1/cards/5efcab153d4408623d3a0481':
+        return MockResponse([{'id': '5efcab153d4408623d3a0484', 'name': 'Test task4', 'desc': 'Test in in-progress', 'due': None, 'dateLastActivity': '2020-07-02T12:59:12.694Z'}], 200)
+
+    return MockResponse(None, 404)
+
+def mock_delete_requests(*args, **kwargs):
+    class MockResponse:
+        def __init__(self, json_data, status_code):
+            self.json_data = json_data
+            self.status_code = status_code
+
+        def json(self):
+            return self.json_data
+
+    if args[0] == 'https://api.trello.com/1/cards/5efcab153d4408623d3a0481':
+        return MockResponse([{'id': '5efcab153d4408623d3a0484', 'name': 'Test task4', 'desc': 'Test in in-progress', 'due': None, 'dateLastActivity': '2020-07-02T12:59:12.694Z'}], 200)
+
+    return MockResponse(None, 404)
+
+
+@mock.patch('requests.get', side_effect=mock_get_requests)
+def test_default_page(mock_get_requests, client):
+    #given
+
+    # when
+    response = client.get("/")
+    response_html = response.data.decode()
+
+    # then 
+    assert response.status_code == 200
+    assert "<title>To-Do App</title>" in response_html
+    assert "<h2>Items</h2>" in response_html
+    assert "Test task1" in response_html
+    assert "Test task2" in response_html
+    assert "Test task3" in response_html
+    assert "Test task4" in response_html
+    assert "Test task5" in response_html
+    assert "Test task6" in response_html
+    assert "Test task7" in response_html
+    assert "Test task8" in response_html
+    assert "Test task9" in response_html
+
+@mock.patch('requests.get', side_effect=mock_get_requests)
+@mock.patch('requests.put', side_effect=mock_put_requests)
+def test_update_page(mock_get_requests, mock_put_requests, client):
+    #given
+
+    # when
+    response = client.post("/update/5efcab153d4408623d3a0481/InProgress", follow_redirects=False)
+    response_html = response.data.decode()
+
+    # then 
+    assert response.status_code == 302
+    assert "<h1>Redirecting...</h1>" in response_html
+
+def test_add_get_page(client):
+    #given
+
+    # when
+    response = client.get("/add", )
+    response_html = response.data.decode()
+
+    # then 
+    assert response.status_code == 200
+    assert "<title>To-Do App</title>" in response_html
+    assert "<h2>Add New Items</h2>" in response_html
+
+@mock.patch('requests.post', side_effect=mock_post_requests)
+def test_add_post_page(mock_post_requests, client):
```

Comment:
I believe this test currently fails because of problems with the mock setup.

Reply:
Also, the test name could be better. What does "add post page" mean? There's no page called "add post".  You're definitely not posting a page.

It should be something more like test_add_todo_item. Or if you want to name it in terms of endpoint and method, it should be _endpoint rather than _page, and I'd normally put method (POST) before the path (/add) because it's more like an English sentence - you're posting to the "add" endpoint.

Reply:
These tests are badly named, they should all be "endpoint" and not "page".

The "add" endpoint has been overloaded to have both a get and a post. This test testing the post REST call.

The reason for it failing was that is was using the wrong .env file. 
No idea why it was working in VS Code for me, but when I check it out to a new directory if failed in the same way it did for you. I've now re-worked and fixed the loading of the .env files.